securedrop_client/queue.py

```diff
@@ -64,21 +66,26 @@ def _process(self, exit_loop: bool) -> None:
                 # priorities are processed in the order that they were submitted
                 # _by the user_ to the queue.
                 self.queue.put_nowait((priority, job))
+                self.pause()
             except ApiInaccessibleError:
                 # This is a guard against #397, we should pause the queue execution when this
                 # happens in the future and flag the situation to the user (see ticket #379).
                 logger.error('Client is not authenticated, skipping job...')
             finally:
                 session.close()
 
-            if exit_loop:
-                return
+    def pause(self):
+        self.queue.enqueue(PauseQueueJob())
+
+    def resume(self):
+        self.process()
 
 
 class ApiJobQueue(QObject):
     # These are the priorities for processing jobs.
     # Lower numbers corresponds to a higher priority.
     JOB_PRIORITIES = {
+        PauseQueueJob: 11,
```

Comment:
Ah, yes, I like how you broke logout down into the two pieces: TokenInvalidationJob (the api piece) and signing the user out in the client. 

> 3. User logs out. UI is in the logged out state immediately regardless of what the queue or network is doing.

This makes sense to me. Just to expand on this... Once the network-health-check functionality is implemented, the client will be able to auto-resume a queue after network issues. This means that we could allow the queues to resume even when the user is logged out. So basically it's what you said: the queues continue operating whether or not the user is logged in. But since the auth token will be invalid, the queues will pause once or if they hit a job that receives an invalid token error. :/ So I think this means only PauseQueueJob will get processed when signed out. Is that your understanding too?

> 2. Client loses network connectivity, queue will pause (aside but related: we should also have the queue pause if it ever hits an auth error because it means the user token expired or is invalidated)

Created an Issue for this aside: https://github.com/freedomofpress/securedrop-client/issues/517